### ISSUE TITLE
Support field-based analysis

### ISSUE THREAD

**Issue by yeikel** [zouzias/spark-lucenerdd#163]
In Lucene , we have the ability to analyze different fields using different analyzers. As initiated in #156  , next natural step should be to support different analysis depending on the field. 

Example using Lucene Core : 

 ```
 Map<String, Analyzer> analyzerPerField = new HashMap<>();
 analyzerPerField.put("firstname", new StandardAnalyzer());
 analyzerPerField.put("address", new WhitespaceAnalyzer());

PerFieldAnalyzerWrapper aWrapper =  new PerFieldAnalyzerWrapper(new StandardAnalyzer(), analyzerPerField);

IndexWriterConfig config = new IndexWriterConfig(aWrapper);
```

More information : [PerFieldAnalyzerWrapper](https://lucene.apache.org/core/8_0_0/analyzers-common/org/apache/lucene/analysis/miscellaneous/PerFieldAnalyzerWrapper.html)

**Comment by zouzias** [comment]
This feature request is valid.

Do you have any proposal on how the API will look like?

As a first draft, I think about passing a `Map[String,String]` where keys are field names and values are the class name of the analyzer, ie. `org.apache.lucene.analyzer.EnglishAnalyzer`. Note: values are string so that they are serializable. WDYT?

**Comment by yeikel** [comment]
> This feature request is valid.
> 
> Do you have any proposal on how the API will look like?
> 
> As a first draft, I think about passing a `Map[String,String]` where keys are field names and values are the class name of the analyzer, ie. `org.apache.lucene.analyzer.EnglishAnalyzer`. Note: values are string so that they are serializable. WDYT?

I think this is more than enough

**Comment by zouzias** [comment]
It it currently WIP: https://github.com/zouzias/spark-lucenerdd/pull/164

**Comment by yeikel** [comment]
@zouzias What support do you need? / What is missing so far?

**Comment by zouzias** [comment]
I need to add a few more unit tests on this and then release it.

**Comment by zouzias** [comment]
@yeikel I just released it under `0.3.7-SNAPSHOT`. The API on block linkage changed a bit but the rest should be backwards compatible. Let me know if you have any feedback.

**Comment by zouzias** [comment]
To be released in 0.3.7, closing.
